rawkit/raw.py

```diff
@@ -175,6 +175,57 @@ def save_thumb(self, filename=None):
         self.libraw.libraw_dcraw_thumb_writer(
             self.data, filename.encode('ascii'))
 
+    def get_4_col_raw(self):
+        """
+        Read the 4 colour raw data
+        The returned numpy array is (given colour RGGB)
+        ---------
+        R B R B ... 
+        G B G B ...
+        R B R B ....
+        . . . . .
+        . . . .  .
+        . . . .   .
+        --------
+
+
+        Returns:
+            array: 4 colour data of the image in unit16. (width x height)
+            str : colour channel description (ie. RGGB, RGBG)
+        """
+        # Unpack the data, so that rawdata is populated
+        self.unpack()
+        rawdata = self.data.contents.rawdata
+
+        # Return None if the image isn't 4 colour, which will happen for some
+        # cameras
+
+        try:
+            rawdata.color4_image.contents
+        except:
+            return None
+
+        # Get image size
+        iheight = rawdata.sizes.iheight
+        iwidth = rawdata.sizes.iwidth
+
+        # Make pointer to data
+
+        data_pointer = ctypes.cast(
+            rawdata.color4_image.contents,
+            ctypes.POINTER(ctypes.c_ushort)
+        )
+
+        # make 2D list
+        data = [[0 for i in range(iheight)] for j in range(iwidth)]
```

Comment:
Might be cleaner, no idea if there's a "standard" way to do this in Python:

```python
data = [None] * iheight * iwidth
```

It might be a pre-mature optimization either way though.